fontbe/src/name.rs

```diff
@@ -42,11 +48,25 @@ impl Work<Context, AnyWorkId, Error> for NameWork {
                 language_id: key.lang_id,
                 string: OffsetMarker::new(value.clone()),
             })
-            .collect::<Vec<_>>();
+            .collect();
+
+        if let Some(name) = extra_tables
+            .as_ref()
+            .and_then(|tables| tables.name.as_ref())
+        {
+            log::info!("merging name records from FEA");
+            merge_name_records(&mut name_records, name);
+        }
 
-        context
-            .name
-            .set(Name::new(name_records.into_iter().collect()));
+        context.name.set(Name::new(name_records));
         Ok(())
     }
 }
+
+fn merge_name_records(records: &mut BTreeSet<NameRecord>, fea_names: &Name) {
```

Comment:
I think we're just overwriting not meaningfully merging so perhaps upsert_name_records would be more descriptive?

Reply:
this is as separate function because I am unclear of what the logic should actually be, and the name was chosen to reflect the semantics, not the specifics of the current implementation.